packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -782,6 +816,107 @@ export class Combobox
     this.updateActiveItemIndex(targetIndex);
   }
 
+  private hideChip(chipEl: HTMLCalciteChipElement): void {
+    chipEl.classList.add(CSS.chipInvisible);
+  }
+
+  private showChip(chipEl: HTMLCalciteChipElement): void {
+    chipEl.classList.remove(CSS.chipInvisible);
+  }
+
+  private maxCompactBreakpoint: number;
+
+  @State() compactDisplayMode = false;
+
+  private refreshDisplayMode = async () => {
+    await componentLoaded(this);
+
+    if (isSingleLike(this.selectionMode)) {
```

Comment:
Let's create a perf follow-up issue for this. I see some opportunities to optimize this since it is an expensive call that gets called frequently:

1. Observe/Unobserve only when `displayMode` is set to `fit-to-line`. 
2. Split DOM reads and writes using `readTask`/`writeTask` (see https://stenciljs.com/docs/api#other)
3. Avoid awaiting on the component to be ready if we know it's single-like or the required input element is not available.
4. Avoid recomputing `inputTextWidth` unless `placeholder` changes.